### ISSUE TITLE
Function prototype call and apply both don't work for a frozen context object

### ISSUE THREAD

**Issue by JasonRammoray** [maxnordlund/function.prototype#3]
Example: 
```js
var obj = {foo: 42};
Object.freeze(obj);
function t() {console.log(this)}
t.call(obj);
```

As a result we get a runtime error (`Uncaught TypeError: object[symbol] is not a function`), because function hasn't been added to a context object, because the latter was frozen.

**Comment by maxnordlund** [comment]
True, but I don't think any JS engine old enough to need this implements `Object.freeze`. I don't have the tweet at hand, but I did ask someone doing research on these old engines, and `try`/`catch` is not a thing. So this needs to be fixed as well, but I didn't get a VM running old Netscape and IE.

**Comment by JasonRammoray** [comment]
True.
Object.freeze has been [shipped to Firefox 4 / IE 9 / Chrome 6 / Opera 12 / Safari 5.1](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze), which are not so old enough browsers compared to Netscape.
Yet I think it is a good idea to test if object is immutable (for example, by adding a property to it and checking if it is there afterwards).
There should be another strategy in case of immutable context.
What do you think?

**Comment by maxnordlund** [comment]
After looking around a bit, and reading the first version of [ECMAScript-262](https://www.ecma-international.org/publications/standards/Ecma-262-arch.htm) it seems like you can use the `typeof` operator to figure out if `Object.freeze` is available.

But according to [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#Specifications) it was first implemented in JavaScript 1.1, which [translates to Netscape 3.0](https://en.wikipedia.org/wiki/JavaScript#Version_history).

Since this project tries to work in those ancient browsers that doesn't work. Both `instanceof` and `try`/`catch` is even younger, introduced in ECMAScript 3.

---

After reading some more, and finding the [changelog for 1.1](https://developer.mozilla.org/en-US/docs/Web/JavaScript/New_in_JavaScript/1.1) I've realized I need 1.1 in order to support `Function.prototype` at all. So the `typeof` solution will work.

**Comment by JasonRammoray** [comment]
Not sure though how inheritance helps to resolve a following case:

```js
var obj = {
    prop: 'value'
};

// Freeze the object by some way

function some (param) {
    return this === obj && param === 42;
}

/*
 * In that case this will point to an object inheriting from obj, but
 * that is not something we need
 */
some.call(obj, 42);
```
